### PR TITLE
fix chosen_nll_loss in chunked losses

### PR DESCRIPTION
**Pull request by kashif**
## Summary
Fix the nll loss in the the chunked loses when the model is a decoder only model, by shifting the logits and targets

- Hardware Type: <BLANK>
- [ ] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Review by shivam15s — APPROVED** [review]
Great catch, and thanks for the quick PR! With the TRL fixes we discussed today, we should be able to closely match the loss curve obtained from TRL Trainer
